edge-hub/core/src/Microsoft.Azure.Devices.Routing.Core/MessageQueueIdHelper.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Routing.Core
+{
+    using System;
+
+    public static class MessageQueueIdHelper
+    {
+        public const string MessageQueueIdDelimiter = "_Pri";
+
+        // The actual ID for the underlying store is of string format: <endpointId>_Pri<priority>
+        // We need to maintain backwards compatibility for existing sequential stores that don't have the "_Pri<x>" suffix.
+        // We use the default priority (2,000,000,000) for this, which means the store ID is just the endpoint ID.
+        public static string GetMessageQueueId(string endpointId, uint priority) => priority == RouteFactory.DefaultPriority ? endpointId : $"{endpointId}{MessageQueueIdDelimiter}{priority}";
+
+        public static (string, uint) ParseMessageQueueId(string messageQueueId)
+        {
+            var idx = messageQueueId.LastIndexOf(MessageQueueIdDelimiter);
+            if (idx < 0)
+            {
+                return (messageQueueId, RouteFactory.DefaultPriority);
+            }
+
+            var endpointId = messageQueueId.Substring(0, idx);
+            var priority = messageQueueId.Substring(idx + MessageQueueIdDelimiter.Length);
+            try
+            {
+                return (endpointId, Convert.ToUInt32(priority));
```

Comment:
nit: why not UInt32.TryParse? exceptions are slow and really for exceptional situations...

Reply:
done